src/psyGen.py

```diff
@@ -1864,16 +2101,31 @@ def iteration_space_arg(self, mapping={}):
                               "we assume there is at least one writer, "
                               "reader/writer, or increment as an argument")
 
-    def set_dependencies(self):
-        for argument in self._args:
-            argument.set_dependencies()
-        # TODO create a summary of dependencies
+    def args_filter(self, arg_types=None, arg_accesses=None):
```

Comment:
You are quite right. args_filter in the loop class uses psyGen.args_filter, not this one. have removed it.